internal/cephfs/controllerserver.go

```diff
@@ -823,6 +819,41 @@ func (cs *ControllerServer) CreateSnapshot(
 	// too.
 	volClient := core.NewSubVolume(parentVolOptions.GetConnection(), &parentVolOptions.SubVolume,
 		parentVolOptions.ClusterID, cs.ClusterName, cs.SetMetadata)
+
+	if parentVolOptions.BackingSnapshot {
+		// update the reference count for parent volume
+		if err = store.UpdateRefCount(ctx, parentVolOptions, cs.ClusterName,
+			cs.SetMetadata, req.GetSecrets()); err != nil {
+			return nil, status.Error(codes.InvalidArgument, "failed to snapshot a snapshot-backed volume")
+		}
+		// Reservation
+		sID, er := store.ReserveSnap(ctx, parentVolOptions, vid.FsSubvolName, cephfsSnap, cr)
+		if er != nil {
+			return nil, status.Error(codes.Internal, er.Error())
+		}
+
+		defer func() {
+			if er != nil {
+				errDefer := store.UndoSnapReservation(ctx, parentVolOptions, *sID, snapName, cr)
+				if errDefer != nil {
+					log.WarningLog(ctx, "failed undoing reservation of snapshot: %s (%s)",
+						requestName, errDefer)
+				}
+			}
+		}()
+
+		// setting creation time as the current time
+		ts := timestamppb.Now()
+		return &csi.CreateSnapshotResponse{
+			Snapshot: &csi.Snapshot{
+				SnapshotId:     sID.SnapshotID,
```

Comment:
As this is a new snapshotID am not sure we can identify the existing snapshot when we create a clone, can we extent the E2E to make sure we are able to create a clone from these snapshots?

Reply:
Having a clone of the backing snapshots, is still remaining to be implemented in controllerserver.

Reply:
why it requires extra work, this is just like other snapshots. if you dont validate the clone how can you ensure its working as expected?

Reply:
Yes, checking on it.

Reply:
Thanks Madhu, updated.
Working now, tested manually by creating a pvc restore from this shallow snapshot and then cloned it.

Reply:
[riyasinghal@fedora ceph-csi]$ sudo kubectl get pvc
NAME                    STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS     AGE
cephfs-pvc-clone-sh     Bound    pvc-47a77e02-946a-45e6-afae-09c2a1744329   1Gi        RWX            csi-cephfs-sc3   7m38s
cephfs-pvc-restore-sh   Bound    pvc-ed3136c3-a7e9-4047-9caa-485ae83cd193   1Gi        RWX            csi-cephfs-sc3   8m54s

Reply:
Hello, could you kindly make the following updates to the E2E test:

1. Create a PVC.
2. Take a snapshot.
3. Create a ROX clone.
4. Take a snapshot of the ROX clone.
5. Remove the PVC created in step 1, the snapshot created in step 2, and the ROX clone created in step 3.
6. Create a new PVC from the snapshot taken in Step 4.
7. Delete all items.

Reply:
Yes, adding a new test with the following steps.

Reply:
Updated, and checked.